pkg/adaptor/cloud/aws/provider.go

```diff
@@ -98,6 +98,25 @@ func (p *awsProvider) CreateInstance(ctx context.Context, podName, sandboxID str
 	//Convert userData to base64
 	userDataEnc := base64.StdEncoding.EncodeToString([]byte(userData))
 
+	// Check if instanceType is empty
+	// If it's not empty, it means it's set via the pod annotation
+	if instanceType == "" {
+		// Set instanceType to default instance type if instanceType is empty
+		instanceType = p.serviceConfig.InstanceType
+	}
+
+	// If instanceTypes is empty and instanceType is not default instance type, return error
+	if len(p.serviceConfig.InstanceTypes) == 0 && instanceType != p.serviceConfig.InstanceType {
+		return nil, fmt.Errorf("instance type %q is not supported.", instanceType)
+	}
+
+	// Check if instanceType is among the supported instance types only if instanceTypes is not empty.
+	if len(p.serviceConfig.InstanceTypes) > 0 {
+		if !util.Contains(p.serviceConfig.InstanceTypes, instanceType) {
+			return nil, fmt.Errorf("instance type %q is not supported", instanceType)
```

Comment:
Do we want to be more lenient here and use the default if an incorrect failure is specified? 
If not could/should that error be added to the pods event list so it is clearer to the end user why provisioning failed?

Reply:
@jtumber-ibm good idea. I think being lenient will be more helpful along with a warning ?